core/src/test/java/io/questdb/cutlass/pgwire/PGFlushQueryCacheTest.java

```diff
@@ -113,6 +123,13 @@ public void testFlushUpdateCache() throws Exception {
     }
 
     private void checkQueryCacheFlushed(long memInitial, long memAfterJoin) {
+        // We need to wait until PG Wire workers process the message. To do so, we simply try to
+        // publish another query flush event. Since we set the queue size to 1, we're able to
+        // publish only when all consumers (PG Wire workers) have processed the previous event.
+        Assert.assertEquals(1, engine.getConfiguration().getQueryCacheEventQueueCapacity());
+        final MPSequence pubSeq = engine.getMessageBus().getQueryCacheEventPubSeq();
```

Comment:
If consumers don't receive the message (e.g. some of them are stuck) won't waitForNext just hang ?

Reply:
> If consumers don't receive the message (e.g. some of them are stuck) won't waitForNext() just hang ?

Yup, that's the case, but we have a timeout rule to catch that, so I wouldn't bother much. If someone changes the code in a way that breaks this assumption, the test would start failing.